provider/openstack/provider.go

```
@@ -719,9 +724,25 @@ func newCredentials(spec environs.CloudSpec) (identity.Credentials, identity.Aut
 		cred.ProjectDomain = credAttrs[CredAttrProjectDomainName]
 		cred.UserDomain = credAttrs[CredAttrUserDomainName]
 		cred.Domain = credAttrs[CredAttrDomainName]
-		authMode = identity.AuthUserPass
-		if cred.Domain != "" || cred.UserDomain != "" || cred.ProjectDomain != "" {
+		if credAttrs[CredAttrVersion] != "" {
+			version, err := strconv.Atoi(credAttrs[CredAttrVersion])
```

We shouldn't ignore the error - an invalid version int should return an error